src/browser/modules/D3Visualization/lib/visualization/renders/init.ts

```diff
@@ -61,51 +62,25 @@ const nodeCaption = new Renderer({
   onGraphChange(selection: any, viz: any) {
     const text = selection
       .selectAll('text.caption')
-      .data((node: any) => node.caption)
+      .data((node: Node) => node.caption)
 
     text
       .enter()
       .append('text')
-      // .classed('caption', true)
+      // Classed element ensures duplicated data will be removed before adding
```

Comment:
@OskarDamkjaer - `selection.exit().remove()` [is typically used to remove “superfluous” elements corresponding to old data](https://github.com/d3/d3-selection/blob/v3.0.0/README.md#selection_exit). That is, we can use this method to remove old nodes' captions when updating nodes drawing, i.e. change nodes' radius. Otherwise, the existing node captions will still here:
<img width="144" alt="Screenshot 2021-12-21 at 00 07 09" src="https://user-images.githubusercontent.com/26452483/146848951-b529c1a6-1f84-4535-b23d-96d9c2f900dd.png">
<img width="195" alt="Screenshot 2021-12-21 at 00 07 20" src="https://user-images.githubusercontent.com/26452483/146848962-d134af1a-f14d-4e06-90bf-97fcc744525a.png">
As you can see the old caption is not removed when updating the node radius. So, those captions were overlapped together.

The mechanism is that we select all elements with the selector `text.caption`. So, `D3` is able to remove old elements and re-render new elements.
https://github.com/neo4j/neo4j-browser/blob/947678af576733051414d286cb16e3fa75ba062b/src/browser/modules/D3Visualization/lib/visualization/renders/init.ts#L60-L86

Previously, we have such code:
https://github.com/neo4j/neo4j-browser/blob/947678af576733051414d286cb16e3fa75ba062b/src/browser/modules/D3Visualization/lib/visualization/renders/init.ts#L92-L117
`nodeIcons` selects all `text` elements and remove old ones when updating, and that's why we don't need explicitly specify class of `nodeCaption`. However, `nodeIcon` is removed in this PR. So, we ran into the problem like the screenshot above. In this case, we need to uncomment the line 69 to set `.classed('caption', true)`.

Reply:
Ah interesting, thanks for explaining!